### ISSUE TITLE
Systemd-logind does not detect activity

### ISSUE THREAD

**Issue by prosoitos** [ch11ng/exwm#537]
My laptop goes to sleep while in full use.

I pinpointed it down to the fact that system-logind does not receive signals that it is not idle and thus applies "IdleAction" after "IdleActionSec" whether it is in use or not.

The only way to stop this happening was to set "IdleAction" to "ignore". This is not really satisfactory as, of course, now it does not go to sleep when it is truly left to idle.

I found this bug report: https://bugs.freedesktop.org/show_bug.cgi?id=77671 which states:

"Your DE needs to inform logind about when it is idle or not, logind cannot figure that out on its own. GNOME and KDE do this correctly. Please file a bug against your DE of choice."

What is surprising is that I had no such issue until December 19. Then the problem started. Of course, I have no clue whether it started because of an update on Arch Linux, on EXWM, or on something else.

Setup info:

OS: Arch Linux (4.20.4-arch1-1-ARCH)
X: xorg-server, version 1.20.3-1
Emacs: GNU Emacs 26.1
EXWM: latest Melpa version

**Comment by ch11ng** [comment]
It seems you'll need a session manager.  But EXWM is not a session manager / DE, neither are other WMs.  You might have to launch EXWM with a session manager, or disable logind and turn to other solutions.

**Comment by prosoitos** [comment]
> EXWM is not a session manager / DE, neither are other WMs

Oh, I see. I hadn't realized. I will install one. Sorry for my confusion between DE and WM. I guess I will close this issue then.

> or disable logind and turn to other solutions

This is what I am doing at the moment (using xautolock to trigger my hybrid-sleep) and it works just fine. But it may be better to have logind running with a DE (?)

Are EXWM users usually using a DE?

**Comment by WJCFerguson** [comment]
> Are EXWM users usually using a DE?

I'm pretty comfortable running it under `[gnome-flashback](https://github.com/WJCFerguson/exwm-gnome-flashback)`.

**Comment by prosoitos** [comment]
Thank you for the suggestion (and your work!). I will check it out. I was also looking at https://github.com/dopsi/console-tdm. But mostly, I am wondering whether it would be better to use a DE at all or to stick with my current setup which doesn't have any. The only downside that I can see if the current issue, but xautolock does the trick to trigger hybrid-sleep and all seems well as long as systemd-logind is disabled.

But I am still a noob in the Arch Linux world and there may be other downsides to my current setup that I am not aware of due to ignorance.

I would appreciate any comment on the necessity (or not) of a DE form people who know better. Thank you!

**Comment by WJCFerguson** [comment]
(My work was little more  than search/replace on someone else's work for `i3`)

I like it to get a fairly well curated set of typical desktop conveniences, with exwm just slotted into it.  I'm not even sure the entire list of features I'm gaining, but it includes: 

* keyboard hotkeys (volume, brightness etc.) with nice graphical notifications
* under Ubuntu, font rendering seems much better under a curated DE - settings I have failed to replicate stand-alone, even with reference to wikis like Arch's
* ssh/gpg agents are properly configured - setting that up caused me headaches
* I also think that it manages, for instance, the auto-mounting of external drives(?)  That works, anyway.

In general, I'm not into recreating a DE so I'm very happy to piggyback on the work already done by others, slotting exwm into it.  Gnome-flashback fits that bill well for me so far.

**Comment by prosoitos** [comment]
Thank you for your answer: it is very helpful.

I got all the things you list to work fine through my current setup. I guess I have embraced the Arch philosophy of building everything yourself lol. So I guess at this point I can go either way then and it seems that it is just a question of preference. But if I hit another issue that a DE would easily fix, I will give it a try. Either way, thank you! This thread helped me clarify things.
